src/watch.rs

```diff
@@ -122,7 +140,7 @@ fn start_event_thread(
             };
 
             if let Ok(event) = in_channel.recv_timeout(timeout) {
-                if is_file_change(event) {
+                if is_file_change(event) && !suspend.load(Ordering::SeqCst) {
```

Comment:
A proper state machine that suspends changes for a given duration might be necessary, but maybe just doing
```rust
if suspend.load(Ordering::SeqCst) {
  sleep(COLLECT_EVENTS_MILLIS)
} else if send_event {
  out_channel.send(...)
}
```
would work as well. (Possibly together with breaking the inner loop if suspend is set.)

Reply:
(Of course, all of this is a bit racy as this might suspend reacting to changes for too long. But that applies to the unwatch solution as well but without any control over how long it takes to re-establish the watches.)

Reply:
Again, I might be missing something, but this does not prevent the "opposite" step to run. The delay I mean is not `COLLECT_EVENTS_MILLIS`, but the one second passed to

```rust
let mut source_watcher = notify::watcher(tx_sources, Duration::from_secs(1))?;
```

So what I think what happens:
1. Switch suspend on
1. Process files, output code
1. Watcher is notified, but delays sending
1. Switch suspend off
1. ... 1sec
1. Watcher sends event

So, as you write, I somehow need to keep suspension for more than the one second of delay.

Reply:
>  So, as you write, I somehow need to keep suspension for more than the one second of delay.

So the only change is using `sleep(Duration::from_secs(2))` in the suspend branch?

Reply:
Ah, sleeping alone won't work as the channel buffers as well. So instead of sleeping, you need to drain the channel for two seconds, e.g.
* deadline = now + 2s
* loop:
* timeout = deadline-now or break
* recv_timeout(timeout)

Reply:
Also as written in the other PR, I don't think you debouncing as you coalesce already, but I think you should use a constant timeout and really wait for quiescence.

Reply:
So, as done in the last commit? It works, and when saving a file multiple times in high frequency, it waits until the last save.